### PR TITLE
Add DirichletAnalytic and Periodic boundary conditions for GRMHD

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

Exactly what the description/title says :shrug: 

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

**Review by fmahebert — DISMISSED** [review]

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/BoundaryConditions/BoundaryCondition.hpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+#include "Domain/BoundaryConditions/BoundaryCondition.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+namespace grmhd::ValenciaDivClean::BoundaryConditions {
+class DirichletAnalytic;
+}  // namespace grmhd::ValenciaDivClean::BoundaryConditions
+/// \endcond
+
+/// \brief Boundary conditions for the scalar wave system
```
Comment (kidder) [inline src/Evolution/Systems/GrMhd/ValenciaDivClean/BoundaryConditions/BoundaryCondition.hpp]: fix system

File: tests/Unit/Evolution/Systems/GrMhd/ValenciaDivClean/BoundaryConditions/Test_Periodic.cpp [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+#include "Domain/BoundaryConditions/Periodic.hpp"
+#include "Evolution/Systems/GrMhd/ValenciaDivClean/BoundaryConditions/BoundaryCondition.hpp"
+#include "Evolution/Systems/GrMhd/ValenciaDivClean/BoundaryConditions/Factory.hpp"
+#include "Framework/TestHelpers.hpp"
+#include "Helpers/Evolution/DiscontinuousGalerkin/BoundaryConditions.hpp"
+
+namespace helpers = TestHelpers::evolution::dg;
+
+SPECTRE_TEST_CASE("Unit.GrMhd.ValenciaDivClean.BoundaryConditions.Periodic",
+                  "[Unit][Burgers]") {
```
Comment (kidder) [inline tests/Unit/Evolution/Systems/GrMhd/ValenciaDivClean/BoundaryConditions/Test_Periodic.cpp]: fix label

**Review by kidder — CHANGES_REQUESTED** [review]
you can squash the minor fixes

**Comment by nilsdeppe** [comment]
Squashed in the fixes. Thanks for the reviews :)

**Review by fmahebert — APPROVED** [review]

**Review by kidder — APPROVED** [review]
